### PR TITLE
Add support for glTF/glb tile content

### PR DESCRIPTION
**Pull request by lilleyse**
This is a WIP implementation of [3DTILES_content_gltf](https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/extensions/3DTILES_content_gltf) targeting the `3d-tiles-next` staging branch.

glTF and glb content is loaded as `Gltf3DTileContent` which is a slimmed down version of `Batched3DModel3DTileContent`. Support for [EXT_feature_metadata](https://github.com/CesiumGS/glTF/tree/feature-metadata/extensions/2.0/Vendor/EXT_feature_metadata) will be in a future PR.

[Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=bVF/b9owFPwqUf4plSYHyLpuK0Ub4YeCSFgh7SiKNAXnAQ8cO40NlEz77nMIW0uLFCv2e3fne2cHJG4S4gpOYphHG6a+UwpSBmIN3Lg1QhP2/eWsR3GIffc+d2s+utLloyvquJ/cdTp5cPpfiAY9xb11AUIvaK38esf2sGpPe9P1IBgtHxNXDX92ct+p4TCIV8N2J/dy78rLFzhw+ulUi/ntO+kmbBnrvRfcffRXerW9qmdXSct2rn0Be3v6Y0Ln17+6k1bncYcZVXV7Lu+Bjvi4Vu8y5X8OzZuQh3wbZcYWYQeZnoLDznDKSR8OtcoFPRwdwVWEHLKLy/8shQwkKE0r+URS4EDSDBNUuAVJojiuhNx4LVv+7HZQkiu/i75hbDL2VWdIiFV+4xSotNqRiqwThtVjal6YAa6sBQu61tEFWUnBQ7NQ+3MZ8qPL0lcuRBKIyhF50jpr+RwQnvWVceU4RVk8ceZyqT0rkXn4jIf7i4zwX7WI0xMxsJe46Fn6QfuA1BLv6eQl9ePuxvxgNqTaM2iWWRrGN0xSkaki1YoOU0GSskjp9GYbutZZUSkLgwW0Yb2mNmLcGhjfhuabZw9Ng7JISt2ZbxgbYw6h2WxYGv+OykQUI18Mt5CxaF/AlrXmoCwSQhqWPp5nKiHYLMreKP8F)

gltf

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @lilleyse!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

File: Source/Scene/Cesium3DTile.js
Comment (ptrgags) [inline Source/Scene/Cesium3DTile.js:939]: Something doesn't sit right with me about breaking the symmetry of the `3DTilesContentFactory` interface. 

What if:

1. You move this `getJsonContent()` function into `3DTilesContentFactory`
2. then any time you call `3DTilesContentFactory.gltf()` or `.json()`, the _factory_ calls `getJsonContent()` just before passing things to the content constructor

This way the factory pattern doesn't change at all, but you still get the benefit of moving the JSON parsing code in its own function.

Comment (ptrgags) [inline Source/Scene/Cesium3DTile.js:939]: Another thing, this function seems very general-purpose, does it make sense to put this in Core since this is just an extended version of `getStringFromTypedArray`?

**Review by ptrgags — COMMENTED** [review]
Code looks good so far, I'll try the sandcastle next.

File: Source/Scene/Model.js
Comment (ptrgags) [inline Source/Scene/Model.js:3834]: @lilleyse oh just a question, do we have any data for testing points?

**Comment by ptrgags** [comment]
I tried the sandcastle (and also tried the `glb` model as well), both are working as far as I can tell!

File: Source/Scene/Cesium3DTile.js
Comment (lilleyse) [reply Source/Scene/Cesium3DTile.js:939]: One problem is it doesn't know whether to call `3DTilesContentFactory.gltf` or `.json` until it's parsed the JSON

File: Source/Scene/Model.js
Comment (lilleyse) [reply Source/Scene/Model.js:3834]: No, we don't have any glTF's with points yet. I don't think it needs to hold up this PR, but it's definitely something we need to do once we unify `PointCloud3DTileContent` and `Model`.

File: Source/Scene/Cesium3DTile.js
Comment (lilleyse) [reply Source/Scene/Cesium3DTile.js:939]: I'll open a PR into `master` that adds `getJsonFromTypedArray`

Comment (ptrgags) [reply Source/Scene/Cesium3DTile.js:939]: Ah I see why you did it this way.

**Review by ptrgags — APPROVED** [review]
Code looks good with the new JSON changes. I can run the sandcastle and see both the glTF and glb versions:

image


image

I'll merge once CI passes

**Comment by ptrgags** [comment]
Merging. Thanks @lilleyse!

**Pull request merged**
